### PR TITLE
feat: support meta and key combination to trigger zoom for scroll-canvas behavior

### PR DESCRIPTION
**Pull request by ttys026**
…vas behavior

##### Checklist

- [x] `npm test` passes
- [ ] tests and/or benchmarks are included
- [x] commit message follows commit guidelines

##### Description of change

- support meta and key combination to trigger zoom for scroll-canvas behavior

### CODE REVIEW

File: packages/site/docs/manual/middle/states/defaultBehavior.en.md [hunk 1]
```diff
@@ -160,7 +160,7 @@ The canvas can be dragged along x direction only.<br /><img src='https://gw.alip
   - `type: 'scroll-canvas'`;
   - `direction`: The direction of dragging that is allowed. Options: `'x'`, `'y'`, `'both'`. `'both'` by default;
   - `enableOptimize`: whether enable optimization, `false` by default. `enableOptimize: true` means hiding all edges and the shapes beside keyShapes of nodes while dragging canvas;
-  - `zoomKey`: switch to zooming while pressing the key and wheeling. Options: `'shift'`, `'ctrl'`, `'alt'`, `'control'`;
+  - `zoomKey`: switch to zooming while pressing the key and wheeling. Options: `'shift'`, `'ctrl'`, `'alt'`, `'control'`, `'meta'`, using an array of these options allows any of these keys to trigger zooming;
```
Comment (cliffordfajardo) [inline packages/site/docs/manual/middle/states/defaultBehavior.en.md:163]: Is the idea behind passing an array to more easily support different operating systems like Mac, Windows, Linux etc?

This seems like a good idea

File: packages/pc/src/behavior/scroll-canvas.ts [hunk 2]
```diff
@@ -25,8 +25,9 @@ export default {
 
   onWheel(ev: IG6GraphEvent) {
     const graph = this.graph;
-    let keyDown = ev[`${this.zoomKey}Key`];
-    if (this.zoomKey === 'control') keyDown = ev.ctrlKey;
+    const zoomKeys = Array.isArray(this.zoomKey) ? this.zoomKey : [this.zoomKey];
+    if (zoomKeys.includes('control')) zoomKeys.push('ctrl');
+    let keyDown = zoomKeys.some(ele => ev[`${ele}Key`]);
     if (keyDown) {
```
Comment (cliffordfajardo) [inline packages/pc/src/behavior/scroll-canvas.ts:31]: @ttys026  - What do you think about making the `zoomKey` value an array to begin with, given the comment below?

> Is the idea behind passing an array to more easily support different operating systems like Mac, Windows, Linux etc which all use different modifier keys? - @cliffordfajardo's [github comment](https://github.com/antvis/G6/pull/3568/files#r821688137)

That simplifies the code? 
For example a user only wants to use one modifier key:
```
zoomKey = ['shift']

// or a user wants to support adding another key (support for another operating system...mac + windows)
zoomKey = [ 'ctrl', 'shift']
```


CC: @Yanyan-Wang

File: packages/site/docs/manual/middle/states/defaultBehavior.en.md (hunk 1 above)
Comment (ttys026) [reply packages/site/docs/manual/middle/states/defaultBehavior.en.md:163]: You're right, that is the original intention of this commit. MacOS command key is interpreted as meta. Pass in an array of zoomKey can help user to support cross-platform zoom behavior

File: packages/pc/src/behavior/scroll-canvas.ts (hunk 2 above)
Comment (ttys026) [reply packages/pc/src/behavior/scroll-canvas.ts:31]: The subjected typescript definition and docs are changed. If we need more demonstrations, we can update the example to show how to use an array of zoom keys

File: packages/pc/src/behavior/scroll-canvas.ts [hunk 3]
```diff
@@ -25,8 +25,9 @@ export default {
 
   onWheel(ev: IG6GraphEvent) {
     const graph = this.graph;
-    let keyDown = ev[`${this.zoomKey}Key`];
-    if (this.zoomKey === 'control') keyDown = ev.ctrlKey;
+    const zoomKeys = Array.isArray(this.zoomKey) ? this.zoomKey : [this.zoomKey];
+    if (zoomKeys.includes('control')) zoomKeys.push('ctrl');
```
Comment (Yanyan-Wang) [inline packages/pc/src/behavior/scroll-canvas.ts:29]: Every onWheel event will push 'ctrl' into zoomKeys, leads to large amount of duplicated 'ctrl' in zoomKeys

**Comment by Yanyan-Wang** [comment]
Thanks for your Pull Request!

**Review by Yanyan-Wang — APPROVED** [review]

**Pull request merged**